modules/articles/client/config/articles.client.routes.js

```diff
@@ -12,22 +12,57 @@ angular.module('articles').config(['$stateProvider',
       })
       .state('articles.list', {
         url: '',
-        templateUrl: 'modules/articles/client/views/list-articles.client.view.html'
+        templateUrl: 'modules/articles/client/views/list-articles.client.view.html',
+        controller: 'ArticlesListController',
+        resolve:{
+          articles: function (Articles, $stateParams, $state) {
+            return Articles.query().$promise;
+          }
+        }
       })
       .state('articles.create', {
         url: '/create',
         templateUrl: 'modules/articles/client/views/create-article.client.view.html',
+        controller: 'ArticlesController',
         data: {
           roles: ['user', 'admin']
+        },
+        resolve: {
+          article: function () {
+            return { };
+          }
         }
       })
       .state('articles.view', {
         url: '/:articleId',
-        templateUrl: 'modules/articles/client/views/view-article.client.view.html'
+        templateUrl: 'modules/articles/client/views/view-article.client.view.html',
+        controller: 'ArticlesController',
+        resolve: {
+          article: function (Articles, $stateParams, $state) {
+            return Articles.get({
+              articleId: $stateParams.articleId
+            }).$promise;
+          }
+        }
       })
       .state('articles.edit', {
         url: '/:articleId/edit',
         templateUrl: 'modules/articles/client/views/edit-article.client.view.html',
+        controller: 'ArticlesController',
+        resolve: {
+          article: function (Articles, Authentication, $stateParams, $state) {
+            return Articles.get({
+              articleId: $stateParams.articleId
+            }).$promise.then(function (article) {
+                //Auth Check
+                if (article.user._id !== Authentication.user._id && Authentication.user.roles.indexOf('admin') === -1) {
+                  //TODO  Change to unauthorized when that PR is merged
+                  $state.go('unauthorized');
+                }
+                return article;
+              });
+          }
+        },
```

Comment:
Because we are not hitting the backend until the actual update of the article.  So the article is loaded in scope when it is viewed.  Now when you put /edit in the URL you will get to the edit route because we are not hitting the backend.  If you tried to update you would get authorization error but that does not prevent you from getting to the route.  This will.